brick_connect_cloud_firestore_build/lib/src/connect_cloud_firestore_model_dictionary_generator.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:brick_build/src/model_dictionary_generator.dart';
+
+class ConnectCloudFirestoreModelDictionaryGenerator extends ModelDictionaryGenerator {
+  @override
+  final requiredImports = """
+// ignore: unused_import
+import 'dart:convert';
+import 'package:brick_cloud_firestore/cloud_firestore.dart' show CloudFirestoreModel, CloudFirestoreAdapter, CloudFirestoreModelDictionary, CloudFirestoreProvider;
+// ignore: unused_import, unused_shown_name
+import 'package:brick_core/core.dart';""";
+
+  /// All classes annotated with `@ConnectCloudFirestore`
+  const ConnectCloudFirestoreModelDictionaryGenerator();
+
+  @override
+  String generate(Map<String, String> classNamesToFileNames) {
+    final adapters = adaptersFromFiles(classNamesToFileNames);
+    final dictionary = dictionaryFromFiles(classNamesToFileNames);
+    final models = modelsFromFiles(classNamesToFileNames);
+    return '''
+${ModelDictionaryGenerator.HEADER}
+$requiredImports
+
+$models
+
+$adapters
+
+/// Cloud Firestore mappings should only be used when initializing a [CloudFirestore]
+final Map<Type, CloudFirestoreAdapter<CloudFirestoreModel>> cloudFirestoreMappings = {
+  $dictionary
+};
+final cloudFirestoreModelDictionary = CloudFirestoreModelDictionary(cloudFirestoreMappings);
+
+class CloudFirestoreRepository extends SingleProviderRepository<CloudFirestoreModel> {
```

Comment:
The end implementation will have to re-implement this class if it chooses to access the repository as a singleton. Also, if the end implementation uses a custom base class, this will again have to be reimplemented. 

By including this repository here, the adapters (as `part` files) are now aware of the implementation of the repository. Which is only a bad thing if the adapters rely on it, which they don't, but it does tempt that bad practice.

Reply:
However, if this were an `abstract class`, it would require the end implementation use its own repository and not rely full on this one.